File: patito/java/1031.java
// Problem ID: 1031
// Title: Mediana

import java.util.Arrays;
import java.util.Scanner;
public class Main
{
	public static void main(String[] args) {
	    int n;
	    Scanner sc = new Scanner(System.in);
	    while(sc.hasNext()){
	        n = sc.nextInt();
	        int num[] = new int[n];
	        for(int i = 0; i <n; i++){
	            num[i] = sc.nextInt();
	        }
	        Arrays.sort(num);
	        if( n % 2 == 0 ){
	            System.out.println("-1");
	        }else{
	            int pos = (int)(n/2);
	            if( pos == 0 ){
	                System.out.println(num[pos]);
	            }else{
	                if(num[pos] != num[pos-1] && num[pos] != num[pos+1]){
	                    System.out.println(num[pos]);
	                }else{
	                    System.out.println("-1");
	                }
	            }
	        }
	    }
	}
}

File: patito/python/1031.py
# Problem ID: 1031
# Title: Mediana

import sys
for linea in sys.stdin:
    if linea == "\n":
        break
    n = int(linea)
    num = list(map(int, input().split()))
    num = num[0:n]
    num.sort()
    if n % 2 == 0:
        print(-1)
    else:
        pos = n//2
        if (num[pos] != num[pos-1] and num[pos] != num[pos+1]) or pos == 0:
            print(num[pos])
        else:
            print(-1)